{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "2f149aed",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "6f9a5267",
   "metadata": {},
   "source": [
    "# GO BP"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "593d65a8",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>consensus independent component 1</th>\n",
       "      <th>consensus independent component 2</th>\n",
       "      <th>consensus independent component 3</th>\n",
       "      <th>consensus independent component 4</th>\n",
       "      <th>consensus independent component 5</th>\n",
       "      <th>consensus independent component 6</th>\n",
       "      <th>consensus independent component 7</th>\n",
       "      <th>consensus independent component 8</th>\n",
       "      <th>consensus independent component 9</th>\n",
       "      <th>consensus independent component 10</th>\n",
       "      <th>...</th>\n",
       "      <th>consensus independent component 106</th>\n",
       "      <th>consensus independent component 107</th>\n",
       "      <th>consensus independent component 108</th>\n",
       "      <th>consensus independent component 109</th>\n",
       "      <th>consensus independent component 110</th>\n",
       "      <th>consensus independent component 111</th>\n",
       "      <th>consensus independent component 112</th>\n",
       "      <th>consensus independent component 113</th>\n",
       "      <th>gene_set_name</th>\n",
       "      <th>Link</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>-2.782916e-16</td>\n",
       "      <td>1.391458e-16</td>\n",
       "      <td>-2.782916e-16</td>\n",
       "      <td>-2.782916e-16</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>-1.391458e-16</td>\n",
       "      <td>1.768696e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>-1.391458e-16</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>1.429627e+00</td>\n",
       "      <td>-1.478641</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>GOBP_MITOCHONDRIAL_GENOME_MAINTENANCE</td>\n",
       "      <td>http://www.gsea-msigdb.org/gsea/msigdb/cards/...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>-1.391458e-16</td>\n",
       "      <td>1.505294e+00</td>\n",
       "      <td>1.391458e-16</td>\n",
       "      <td>1.391458e-16</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>-1.391458e-16</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>1.391458e-16</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>-1.391458e-16</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>GOBP_SINGLE_STRAND_BREAK_REPAIR</td>\n",
       "      <td>http://www.gsea-msigdb.org/gsea/msigdb/cards/...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>-5.668647e-01</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>2.442695e+00</td>\n",
       "      <td>4.174375e-16</td>\n",
       "      <td>2.078227e+00</td>\n",
       "      <td>-1.187087e+00</td>\n",
       "      <td>-1.872795e+00</td>\n",
       "      <td>9.941600e-01</td>\n",
       "      <td>1.610233e+00</td>\n",
       "      <td>...</td>\n",
       "      <td>0.482672</td>\n",
       "      <td>-2.782916e-16</td>\n",
       "      <td>1.391458e-16</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>-5.963215e-01</td>\n",
       "      <td>-1.223743e+00</td>\n",
       "      <td>-0.651827</td>\n",
       "      <td>2.530474e+00</td>\n",
       "      <td>GOBP_REGULATION_OF_DNA_RECOMBINATION</td>\n",
       "      <td>http://www.gsea-msigdb.org/gsea/msigdb/cards/...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1.391458e-16</td>\n",
       "      <td>1.391458e-16</td>\n",
       "      <td>1.391458e-16</td>\n",
       "      <td>-1.391458e-16</td>\n",
       "      <td>-2.782916e-16</td>\n",
       "      <td>-1.391458e-16</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>-1.391458e-16</td>\n",
       "      <td>-1.246121e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>-1.391458e-16</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>-1.391458e-16</td>\n",
       "      <td>1.391458e-16</td>\n",
       "      <td>-1.391458e-16</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>-1.391458e-16</td>\n",
       "      <td>GOBP_RIBOSOMAL_LARGE_SUBUNIT_ASSEMBLY</td>\n",
       "      <td>http://www.gsea-msigdb.org/gsea/msigdb/cards/...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>-1.391458e-16</td>\n",
       "      <td>1.391458e-16</td>\n",
       "      <td>2.782916e-16</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>1.391458e-16</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>-1.391458e-16</td>\n",
       "      <td>1.391458e-16</td>\n",
       "      <td>-1.391458e-16</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>-1.391458e-16</td>\n",
       "      <td>GOBP_RIBOSOMAL_SMALL_SUBUNIT_ASSEMBLY</td>\n",
       "      <td>http://www.gsea-msigdb.org/gsea/msigdb/cards/...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows Ã— 115 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "   consensus independent component 1  consensus independent component 2  \\\n",
       "0                      -2.782916e-16                       1.391458e-16   \n",
       "1                      -1.391458e-16                       1.505294e+00   \n",
       "2                      -5.668647e-01                       0.000000e+00   \n",
       "3                       1.391458e-16                       1.391458e-16   \n",
       "4                       0.000000e+00                       0.000000e+00   \n",
       "\n",
       "   consensus independent component 3  consensus independent component 4  \\\n",
       "0                      -2.782916e-16                      -2.782916e-16   \n",
       "1                       1.391458e-16                       1.391458e-16   \n",
       "2                       0.000000e+00                       2.442695e+00   \n",
       "3                       1.391458e-16                      -1.391458e-16   \n",
       "4                       0.000000e+00                      -1.391458e-16   \n",
       "\n",
       "   consensus independent component 5  consensus independent component 6  \\\n",
       "0                       0.000000e+00                       0.000000e+00   \n",
       "1                       0.000000e+00                       0.000000e+00   \n",
       "2                       4.174375e-16                       2.078227e+00   \n",
       "3                      -2.782916e-16                      -1.391458e-16   \n",
       "4                       1.391458e-16                       2.782916e-16   \n",
       "\n",
       "   consensus independent component 7  consensus independent component 8  \\\n",
       "0                       0.000000e+00                      -1.391458e-16   \n",
       "1                      -1.391458e-16                       0.000000e+00   \n",
       "2                      -1.187087e+00                      -1.872795e+00   \n",
       "3                       0.000000e+00                      -1.391458e-16   \n",
       "4                       0.000000e+00                       0.000000e+00   \n",
       "\n",
       "   consensus independent component 9  consensus independent component 10  ...  \\\n",
       "0                       1.768696e+00                        0.000000e+00  ...   \n",
       "1                       0.000000e+00                        1.391458e-16  ...   \n",
       "2                       9.941600e-01                        1.610233e+00  ...   \n",
       "3                      -1.246121e+00                        0.000000e+00  ...   \n",
       "4                       1.391458e-16                        0.000000e+00  ...   \n",
       "\n",
       "   consensus independent component 106  consensus independent component 107  \\\n",
       "0                             0.000000                        -1.391458e-16   \n",
       "1                             0.000000                         0.000000e+00   \n",
       "2                             0.482672                        -2.782916e-16   \n",
       "3                             0.000000                        -1.391458e-16   \n",
       "4                             0.000000                         0.000000e+00   \n",
       "\n",
       "   consensus independent component 108  consensus independent component 109  \\\n",
       "0                         0.000000e+00                         0.000000e+00   \n",
       "1                         0.000000e+00                        -1.391458e-16   \n",
       "2                         1.391458e-16                         0.000000e+00   \n",
       "3                         0.000000e+00                        -1.391458e-16   \n",
       "4                        -1.391458e-16                         1.391458e-16   \n",
       "\n",
       "   consensus independent component 110  consensus independent component 111  \\\n",
       "0                         0.000000e+00                         1.429627e+00   \n",
       "1                         0.000000e+00                         0.000000e+00   \n",
       "2                        -5.963215e-01                        -1.223743e+00   \n",
       "3                         1.391458e-16                        -1.391458e-16   \n",
       "4                        -1.391458e-16                         0.000000e+00   \n",
       "\n",
       "   consensus independent component 112  consensus independent component 113  \\\n",
       "0                            -1.478641                         0.000000e+00   \n",
       "1                             0.000000                         0.000000e+00   \n",
       "2                            -0.651827                         2.530474e+00   \n",
       "3                             0.000000                        -1.391458e-16   \n",
       "4                             0.000000                        -1.391458e-16   \n",
       "\n",
       "                           gene_set_name  \\\n",
       "0  GOBP_MITOCHONDRIAL_GENOME_MAINTENANCE   \n",
       "1        GOBP_SINGLE_STRAND_BREAK_REPAIR   \n",
       "2   GOBP_REGULATION_OF_DNA_RECOMBINATION   \n",
       "3  GOBP_RIBOSOMAL_LARGE_SUBUNIT_ASSEMBLY   \n",
       "4  GOBP_RIBOSOMAL_SMALL_SUBUNIT_ASSEMBLY   \n",
       "\n",
       "                                                Link  \n",
       "0   http://www.gsea-msigdb.org/gsea/msigdb/cards/...  \n",
       "1   http://www.gsea-msigdb.org/gsea/msigdb/cards/...  \n",
       "2   http://www.gsea-msigdb.org/gsea/msigdb/cards/...  \n",
       "3   http://www.gsea-msigdb.org/gsea/msigdb/cards/...  \n",
       "4   http://www.gsea-msigdb.org/gsea/msigdb/cards/...  \n",
       "\n",
       "[5 rows x 115 columns]"
      ]
     },
     "execution_count": 2,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Read the GSEA outcome\n",
    "REACTOME = False\n",
    "if REACTOME:\n",
    "    link = '/data/g0017139/CNA/enrichment_matrix_Reactome.tsv'\n",
    "else:\n",
    "    link = '/data/g0017139/CNA/enrichment_matrix_Gene Ontology - Biolocal Processes.tsv'\n",
    "\n",
    "df = pd.read_csv(link,sep='\\t')\n",
    "df[['gene_set_name', 'Link']] = df['Unnamed: 0'].str.split(' --',expand=True)\n",
    "df = df.drop('Unnamed: 0', axis=1)\n",
    "df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "781f4d94",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene_set_name</th>\n",
       "      <th>is_immune_process</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>REACTOME_2_LTR_CIRCLE_FORMATION</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>REACTOME_A_TETRASACCHARIDE_LINKER_SEQUENCE_IS_...</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>REACTOME_ABACAVIR_METABOLISM</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>REACTOME_ABACAVIR_TRANSMEMBRANE_TRANSPORT</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>REACTOME_ABACAVIR_TRANSPORT_AND_METABOLISM</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                       gene_set_name  is_immune_process\n",
       "0                    REACTOME_2_LTR_CIRCLE_FORMATION              False\n",
       "1  REACTOME_A_TETRASACCHARIDE_LINKER_SEQUENCE_IS_...              False\n",
       "2                       REACTOME_ABACAVIR_METABOLISM              False\n",
       "3          REACTOME_ABACAVIR_TRANSMEMBRANE_TRANSPORT              False\n",
       "4         REACTOME_ABACAVIR_TRANSPORT_AND_METABOLISM              False"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "test = pd.read_csv('/data/g0017139/CNA/reactome_immune_gene_sets.tsv', sep='\\t')\n",
    "test.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "0f8778ed",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene_set_name</th>\n",
       "      <th>in_manual_selection</th>\n",
       "      <th>in_immune_terms</th>\n",
       "      <th>is_immune_process</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>GOBP_10_FORMYLTETRAHYDROFOLATE_METABOLIC_PROCESS</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>GOBP_2_OXOGLUTARATE_METABOLIC_PROCESS</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>GOBP_2FE_2S_CLUSTER_ASSEMBLY</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>GOBP_3_PHOSPHOADENOSINE_5_PHOSPHOSULFATE_BIOSY...</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>GOBP_3_PHOSPHOADENOSINE_5_PHOSPHOSULFATE_METAB...</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                       gene_set_name  in_manual_selection  \\\n",
       "0   GOBP_10_FORMYLTETRAHYDROFOLATE_METABOLIC_PROCESS                False   \n",
       "1              GOBP_2_OXOGLUTARATE_METABOLIC_PROCESS                False   \n",
       "2                       GOBP_2FE_2S_CLUSTER_ASSEMBLY                False   \n",
       "3  GOBP_3_PHOSPHOADENOSINE_5_PHOSPHOSULFATE_BIOSY...                False   \n",
       "4  GOBP_3_PHOSPHOADENOSINE_5_PHOSPHOSULFATE_METAB...                False   \n",
       "\n",
       "   in_immune_terms  is_immune_process  \n",
       "0            False              False  \n",
       "1            False              False  \n",
       "2            False              False  \n",
       "3            False              False  \n",
       "4            False              False  "
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "test1 = pd.read_csv('/data/g0017139/CNA/gobp_immune_gene_sets_manualselection_updated.tsv', sep='\\t')\n",
    "test1.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "1c2dc2bf",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(7481, 4)"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "test1.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "732a004f",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(1604, 2)"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "test.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "5dc77f25",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "9085"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "1604 + 7481"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "0cc5928f",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene_set_name</th>\n",
       "      <th>is_immune_process</th>\n",
       "      <th>in_manual_selection</th>\n",
       "      <th>in_immune_terms</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>REACTOME_2_LTR_CIRCLE_FORMATION</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>REACTOME_A_TETRASACCHARIDE_LINKER_SEQUENCE_IS_...</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>REACTOME_ABACAVIR_METABOLISM</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>REACTOME_ABACAVIR_TRANSMEMBRANE_TRANSPORT</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>REACTOME_ABACAVIR_TRANSPORT_AND_METABOLISM</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7476</th>\n",
       "      <td>GOBP_ZINC_ION_HOMEOSTASIS</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7477</th>\n",
       "      <td>GOBP_ZINC_ION_IMPORT_ACROSS_PLASMA_MEMBRANE</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7478</th>\n",
       "      <td>GOBP_ZINC_ION_TRANSPORT</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7479</th>\n",
       "      <td>GOBP_ZYMOGEN_ACTIVATION</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7480</th>\n",
       "      <td>GOBP_ZYMOGEN_INHIBITION</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>9085 rows Ã— 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                          gene_set_name  is_immune_process  \\\n",
       "0                       REACTOME_2_LTR_CIRCLE_FORMATION              False   \n",
       "1     REACTOME_A_TETRASACCHARIDE_LINKER_SEQUENCE_IS_...              False   \n",
       "2                          REACTOME_ABACAVIR_METABOLISM              False   \n",
       "3             REACTOME_ABACAVIR_TRANSMEMBRANE_TRANSPORT              False   \n",
       "4            REACTOME_ABACAVIR_TRANSPORT_AND_METABOLISM              False   \n",
       "...                                                 ...                ...   \n",
       "7476                          GOBP_ZINC_ION_HOMEOSTASIS              False   \n",
       "7477        GOBP_ZINC_ION_IMPORT_ACROSS_PLASMA_MEMBRANE              False   \n",
       "7478                            GOBP_ZINC_ION_TRANSPORT              False   \n",
       "7479                            GOBP_ZYMOGEN_ACTIVATION              False   \n",
       "7480                            GOBP_ZYMOGEN_INHIBITION              False   \n",
       "\n",
       "     in_manual_selection in_immune_terms  \n",
       "0                    NaN             NaN  \n",
       "1                    NaN             NaN  \n",
       "2                    NaN             NaN  \n",
       "3                    NaN             NaN  \n",
       "4                    NaN             NaN  \n",
       "...                  ...             ...  \n",
       "7476               False           False  \n",
       "7477               False           False  \n",
       "7478               False           False  \n",
       "7479               False           False  \n",
       "7480               False           False  \n",
       "\n",
       "[9085 rows x 4 columns]"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pd.concat([test, test1])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "617534b6",
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "# Load the imune set names\n",
    "if REACTOME:\n",
    "    immune_df = pd.read_csv('/data/g0017139/CNA/reactome_immune_gene_sets.tsv', sep='\\t')\n",
    "else:\n",
    "    immune_df = pd.read_csv('/data/g0017139/CNA/gobp_immune_gene_sets_manualselection_updated.tsv', sep='\\t')\n",
    "\n",
    "immune_df = immune_df[immune_df['is_immune_process'] == True]\n",
    "\n",
    "\n",
    "#df = pd.merge(df, immune_df, left_on='gene_set_name', right_on='gene_set_name',how='left')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "bc8fd0ad",
   "metadata": {},
   "source": [
    "From GO-BP, all gene set were selected that contained the motif â€˜METABOLIC_PROCESSâ€™ in the title. In addition, gene sets that contained the name of a metabolite or class of metabolites in combination with the motif â€˜_TRANSPORTâ€™ in the title were selected. Furthermore, gene sets not containing these title motifs, but nevertheless associated with (cancer) metabolism, were manually selected on the basis of metabolic pathway names"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "eefa322d",
   "metadata": {
    "scrolled": false
   },
   "outputs": [],
   "source": [
    "# Load the methabolic names\n",
    "metabolic = pd.read_csv('/data/g0017139/CNA/metabolic_gene_sets.txt', sep='\\t')['GO_NUCLEOBASE_METABOLIC_PROCESS']\n",
    "if not REACTOME:\n",
    "    metabolic = metabolic.str.replace('GO','GOBP')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "7fe75a64",
   "metadata": {},
   "outputs": [],
   "source": [
    "#Load DNA Repair\n",
    "dna_df = pd.read_excel('/data/g0017139/CNA/GOBP_REACTOME_RELEVANT_GENESETS.xlsx',header=None)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "61514e1f",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Merge evertying with the results of df\n",
    "df = pd.merge(df, immune_df, left_on='gene_set_name', right_on='gene_set_name',how='left')\n",
    "df['is_methabolic_process'] = df['gene_set_name'].isin(metabolic)\n",
    "df['is_DNA_Repair'] = df['gene_set_name'].isin(dna_df[0])\n",
    "\n",
    "# Only keep that has at least one True\n",
    "df['keep'] = ((df['is_immune_process']==True) | (df['is_methabolic_process']==True) | (df['is_DNA_Repair']==True))\n",
    "analysis_df = df[df['keep']==True]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "b101e8a2",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "is_methabolic_process = 267\n",
      "is_immune_process = 675\n",
      "is_DNA_Repair = 270\n",
      "To keep = 1204\n",
      "total = 5030\n"
     ]
    }
   ],
   "source": [
    "print(f'is_methabolic_process = {df[\"is_methabolic_process\"].sum()}')\n",
    "print(f'is_immune_process = {df[\"is_immune_process\"].sum()}')\n",
    "print(f'is_DNA_Repair = {df[\"is_DNA_Repair\"].sum()}')\n",
    "print(f'To keep = {df[\"keep\"].sum()}')\n",
    "print(f'total = {len(df)}')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "92cc6681",
   "metadata": {},
   "outputs": [],
   "source": [
    "import scipy.stats as st"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "810559c1",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/data/g0017139/.envs/ClusterEnv/lib/python3.9/site-packages/pandas/core/indexing.py:1667: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  self.obj[key] = value\n"
     ]
    }
   ],
   "source": [
    "# Drop the not relevant genes\n",
    "test_df = df.drop(['gene_set_name','Link','is_immune_process', 'is_methabolic_process','is_DNA_Repair',  'keep'],\n",
    "                 axis=1)\n",
    "shape = test_df.shape\n",
    "# What is the number of tests? \n",
    "# Bonferroni correction\n",
    "value = st.norm.ppf(0.05/(shape[0]*shape[1]))\n",
    "test_df = test_df.abs()\n",
    "test_df = (test_df > abs(value))\n",
    "\n",
    "# See of any value in the row is still significant (So higher than corrected value) and only keep those\n",
    "df['Any_Sig'] = test_df.sum(axis=1) > 0\n",
    "test_df = test_df[test_df.sum(axis=1) > 0]\n",
    "# Loop over the significant rows and get those column names (The names of the components)\n",
    "significant_conponents = {}\n",
    "for index, row in test_df.iterrows():\n",
    "    significant_conponents[index] = list(row[row==True].index)\n",
    "\n",
    "# Only keep the rows that have both a signifant value and this signicant value is in a relevent gene for the gene sets\n",
    "biology_df = df[((df['Any_Sig'] == True) & (df['keep'] == True))]\n",
    "# Add the signifcant components and change Nan to False\n",
    "biology_df.loc[:,\"Significant\"] = pd.Series(significant_conponents)\n",
    "biology_df = biology_df.fillna(False)\n",
    "\n",
    "# Count how often each indipendent components has a gene that is linked to a geneset that is linked to a process\n",
    "counts = {}\n",
    "imp_rows = ['is_immune_process', 'is_methabolic_process', 'is_DNA_Repair']\n",
    "for index, row in biology_df.iterrows():\n",
    "    for comp in row['Significant']:\n",
    "        if comp not in counts:\n",
    "            counts[comp] = {\n",
    "                'is_immune_process': 0,\n",
    "                'is_methabolic_process': 0,\n",
    "                'is_DNA_Repair': 0\n",
    "            }\n",
    "        for imp_row in imp_rows:\n",
    "            counts[comp][imp_row] += row[imp_row]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "613133f4",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'consensus independent component 39': {'is_immune_process': 0,\n",
       "  'is_methabolic_process': 1,\n",
       "  'is_DNA_Repair': 114},\n",
       " 'consensus independent component 12': {'is_immune_process': 138,\n",
       "  'is_methabolic_process': 0,\n",
       "  'is_DNA_Repair': 0},\n",
       " 'consensus independent component 15': {'is_immune_process': 22,\n",
       "  'is_methabolic_process': 0,\n",
       "  'is_DNA_Repair': 0},\n",
       " 'consensus independent component 24': {'is_immune_process': 140,\n",
       "  'is_methabolic_process': 0,\n",
       "  'is_DNA_Repair': 0},\n",
       " 'consensus independent component 67': {'is_immune_process': 9,\n",
       "  'is_methabolic_process': 0,\n",
       "  'is_DNA_Repair': 0},\n",
       " 'consensus independent component 7': {'is_immune_process': 3,\n",
       "  'is_methabolic_process': 0,\n",
       "  'is_DNA_Repair': 0},\n",
       " 'consensus independent component 48': {'is_immune_process': 5,\n",
       "  'is_methabolic_process': 0,\n",
       "  'is_DNA_Repair': 0},\n",
       " 'consensus independent component 97': {'is_immune_process': 13,\n",
       "  'is_methabolic_process': 8,\n",
       "  'is_DNA_Repair': 0},\n",
       " 'consensus independent component 37': {'is_immune_process': 1,\n",
       "  'is_methabolic_process': 0,\n",
       "  'is_DNA_Repair': 0},\n",
       " 'consensus independent component 111': {'is_immune_process': 0,\n",
       "  'is_methabolic_process': 1,\n",
       "  'is_DNA_Repair': 0},\n",
       " 'consensus independent component 63': {'is_immune_process': 1,\n",
       "  'is_methabolic_process': 2,\n",
       "  'is_DNA_Repair': 0},\n",
       " 'consensus independent component 69': {'is_immune_process': 0,\n",
       "  'is_methabolic_process': 1,\n",
       "  'is_DNA_Repair': 0},\n",
       " 'consensus independent component 81': {'is_immune_process': 0,\n",
       "  'is_methabolic_process': 3,\n",
       "  'is_DNA_Repair': 0},\n",
       " 'consensus independent component 10': {'is_immune_process': 3,\n",
       "  'is_methabolic_process': 0,\n",
       "  'is_DNA_Repair': 0},\n",
       " 'consensus independent component 109': {'is_immune_process': 8,\n",
       "  'is_methabolic_process': 0,\n",
       "  'is_DNA_Repair': 0},\n",
       " 'consensus independent component 78': {'is_immune_process': 0,\n",
       "  'is_methabolic_process': 5,\n",
       "  'is_DNA_Repair': 0},\n",
       " 'consensus independent component 49': {'is_immune_process': 2,\n",
       "  'is_methabolic_process': 1,\n",
       "  'is_DNA_Repair': 0},\n",
       " 'consensus independent component 51': {'is_immune_process': 4,\n",
       "  'is_methabolic_process': 0,\n",
       "  'is_DNA_Repair': 0}}"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "counts"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "c0caad32",
   "metadata": {},
   "source": [
    "# Peak analysis"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "be30fb8d",
   "metadata": {},
   "source": [
    "# Leukemia"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "c692657c",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chrom</th>\n",
       "      <th>chromStart</th>\n",
       "      <th>chromEnd</th>\n",
       "      <th>name</th>\n",
       "      <th>extreme_value_region_status</th>\n",
       "      <th>mappings_in_region</th>\n",
       "      <th>hyperparameter</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>3396</th>\n",
       "      <td>6</td>\n",
       "      <td>32814882.0</td>\n",
       "      <td>33734304.0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>36</td>\n",
       "      <td>interval_coverage:10 interval_coverage_CI:0.95...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      chrom  chromStart    chromEnd  name  extreme_value_region_status  \\\n",
       "3396      6  32814882.0  33734304.0     1                            1   \n",
       "\n",
       "      mappings_in_region                                     hyperparameter  \n",
       "3396                  36  interval_coverage:10 interval_coverage_CI:0.95...  "
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df = pd.read_csv('/data/g0017139/CNA/_extreme_valued_regions_all_chromosomes.txt', sep='\\t')\n",
    "df = df[df['extreme_value_region_status'] !=0]\n",
    "df = df[df['mappings_in_region'] >= 10]\n",
    "df['name'] = df['name'].str.split(' ',expand=True)[3].astype(int)\n",
    "df[df['name'] == 1]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "99cb96ee",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "TryTwo",
   "language": "python",
   "name": "trytwo"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.6"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
