{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 169,
   "id": "fc73ef13",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 170,
   "id": "6e858099",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>consensus independent component 1</th>\n",
       "      <th>consensus independent component 2</th>\n",
       "      <th>consensus independent component 3</th>\n",
       "      <th>consensus independent component 4</th>\n",
       "      <th>consensus independent component 5</th>\n",
       "      <th>consensus independent component 6</th>\n",
       "      <th>consensus independent component 7</th>\n",
       "      <th>consensus independent component 8</th>\n",
       "      <th>consensus independent component 9</th>\n",
       "      <th>consensus independent component 10</th>\n",
       "      <th>...</th>\n",
       "      <th>consensus independent component 106</th>\n",
       "      <th>consensus independent component 107</th>\n",
       "      <th>consensus independent component 108</th>\n",
       "      <th>consensus independent component 109</th>\n",
       "      <th>consensus independent component 110</th>\n",
       "      <th>consensus independent component 111</th>\n",
       "      <th>consensus independent component 112</th>\n",
       "      <th>consensus independent component 113</th>\n",
       "      <th>gene_set_name</th>\n",
       "      <th>Link</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>-2.782916e-16</td>\n",
       "      <td>1.391458e-16</td>\n",
       "      <td>-2.782916e-16</td>\n",
       "      <td>-2.782916e-16</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>-1.391458e-16</td>\n",
       "      <td>1.768696e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>-1.391458e-16</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>1.429627e+00</td>\n",
       "      <td>-1.478641</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>GOBP_MITOCHONDRIAL_GENOME_MAINTENANCE</td>\n",
       "      <td>http://www.gsea-msigdb.org/gsea/msigdb/cards/...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>-1.391458e-16</td>\n",
       "      <td>1.505294e+00</td>\n",
       "      <td>1.391458e-16</td>\n",
       "      <td>1.391458e-16</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>-1.391458e-16</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>1.391458e-16</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>-1.391458e-16</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>GOBP_SINGLE_STRAND_BREAK_REPAIR</td>\n",
       "      <td>http://www.gsea-msigdb.org/gsea/msigdb/cards/...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>-5.668647e-01</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>2.442695e+00</td>\n",
       "      <td>4.174375e-16</td>\n",
       "      <td>2.078227e+00</td>\n",
       "      <td>-1.187087e+00</td>\n",
       "      <td>-1.872795e+00</td>\n",
       "      <td>9.941600e-01</td>\n",
       "      <td>1.610233e+00</td>\n",
       "      <td>...</td>\n",
       "      <td>0.482672</td>\n",
       "      <td>-2.782916e-16</td>\n",
       "      <td>1.391458e-16</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>-5.963215e-01</td>\n",
       "      <td>-1.223743e+00</td>\n",
       "      <td>-0.651827</td>\n",
       "      <td>2.530474e+00</td>\n",
       "      <td>GOBP_REGULATION_OF_DNA_RECOMBINATION</td>\n",
       "      <td>http://www.gsea-msigdb.org/gsea/msigdb/cards/...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1.391458e-16</td>\n",
       "      <td>1.391458e-16</td>\n",
       "      <td>1.391458e-16</td>\n",
       "      <td>-1.391458e-16</td>\n",
       "      <td>-2.782916e-16</td>\n",
       "      <td>-1.391458e-16</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>-1.391458e-16</td>\n",
       "      <td>-1.246121e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>-1.391458e-16</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>-1.391458e-16</td>\n",
       "      <td>1.391458e-16</td>\n",
       "      <td>-1.391458e-16</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>-1.391458e-16</td>\n",
       "      <td>GOBP_RIBOSOMAL_LARGE_SUBUNIT_ASSEMBLY</td>\n",
       "      <td>http://www.gsea-msigdb.org/gsea/msigdb/cards/...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>-1.391458e-16</td>\n",
       "      <td>1.391458e-16</td>\n",
       "      <td>2.782916e-16</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>1.391458e-16</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>-1.391458e-16</td>\n",
       "      <td>1.391458e-16</td>\n",
       "      <td>-1.391458e-16</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>-1.391458e-16</td>\n",
       "      <td>GOBP_RIBOSOMAL_SMALL_SUBUNIT_ASSEMBLY</td>\n",
       "      <td>http://www.gsea-msigdb.org/gsea/msigdb/cards/...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 115 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "   consensus independent component 1  consensus independent component 2  \\\n",
       "0                      -2.782916e-16                       1.391458e-16   \n",
       "1                      -1.391458e-16                       1.505294e+00   \n",
       "2                      -5.668647e-01                       0.000000e+00   \n",
       "3                       1.391458e-16                       1.391458e-16   \n",
       "4                       0.000000e+00                       0.000000e+00   \n",
       "\n",
       "   consensus independent component 3  consensus independent component 4  \\\n",
       "0                      -2.782916e-16                      -2.782916e-16   \n",
       "1                       1.391458e-16                       1.391458e-16   \n",
       "2                       0.000000e+00                       2.442695e+00   \n",
       "3                       1.391458e-16                      -1.391458e-16   \n",
       "4                       0.000000e+00                      -1.391458e-16   \n",
       "\n",
       "   consensus independent component 5  consensus independent component 6  \\\n",
       "0                       0.000000e+00                       0.000000e+00   \n",
       "1                       0.000000e+00                       0.000000e+00   \n",
       "2                       4.174375e-16                       2.078227e+00   \n",
       "3                      -2.782916e-16                      -1.391458e-16   \n",
       "4                       1.391458e-16                       2.782916e-16   \n",
       "\n",
       "   consensus independent component 7  consensus independent component 8  \\\n",
       "0                       0.000000e+00                      -1.391458e-16   \n",
       "1                      -1.391458e-16                       0.000000e+00   \n",
       "2                      -1.187087e+00                      -1.872795e+00   \n",
       "3                       0.000000e+00                      -1.391458e-16   \n",
       "4                       0.000000e+00                       0.000000e+00   \n",
       "\n",
       "   consensus independent component 9  consensus independent component 10  ...  \\\n",
       "0                       1.768696e+00                        0.000000e+00  ...   \n",
       "1                       0.000000e+00                        1.391458e-16  ...   \n",
       "2                       9.941600e-01                        1.610233e+00  ...   \n",
       "3                      -1.246121e+00                        0.000000e+00  ...   \n",
       "4                       1.391458e-16                        0.000000e+00  ...   \n",
       "\n",
       "   consensus independent component 106  consensus independent component 107  \\\n",
       "0                             0.000000                        -1.391458e-16   \n",
       "1                             0.000000                         0.000000e+00   \n",
       "2                             0.482672                        -2.782916e-16   \n",
       "3                             0.000000                        -1.391458e-16   \n",
       "4                             0.000000                         0.000000e+00   \n",
       "\n",
       "   consensus independent component 108  consensus independent component 109  \\\n",
       "0                         0.000000e+00                         0.000000e+00   \n",
       "1                         0.000000e+00                        -1.391458e-16   \n",
       "2                         1.391458e-16                         0.000000e+00   \n",
       "3                         0.000000e+00                        -1.391458e-16   \n",
       "4                        -1.391458e-16                         1.391458e-16   \n",
       "\n",
       "   consensus independent component 110  consensus independent component 111  \\\n",
       "0                         0.000000e+00                         1.429627e+00   \n",
       "1                         0.000000e+00                         0.000000e+00   \n",
       "2                        -5.963215e-01                        -1.223743e+00   \n",
       "3                         1.391458e-16                        -1.391458e-16   \n",
       "4                        -1.391458e-16                         0.000000e+00   \n",
       "\n",
       "   consensus independent component 112  consensus independent component 113  \\\n",
       "0                            -1.478641                         0.000000e+00   \n",
       "1                             0.000000                         0.000000e+00   \n",
       "2                            -0.651827                         2.530474e+00   \n",
       "3                             0.000000                        -1.391458e-16   \n",
       "4                             0.000000                        -1.391458e-16   \n",
       "\n",
       "                           gene_set_name  \\\n",
       "0  GOBP_MITOCHONDRIAL_GENOME_MAINTENANCE   \n",
       "1        GOBP_SINGLE_STRAND_BREAK_REPAIR   \n",
       "2   GOBP_REGULATION_OF_DNA_RECOMBINATION   \n",
       "3  GOBP_RIBOSOMAL_LARGE_SUBUNIT_ASSEMBLY   \n",
       "4  GOBP_RIBOSOMAL_SMALL_SUBUNIT_ASSEMBLY   \n",
       "\n",
       "                                                Link  \n",
       "0   http://www.gsea-msigdb.org/gsea/msigdb/cards/...  \n",
       "1   http://www.gsea-msigdb.org/gsea/msigdb/cards/...  \n",
       "2   http://www.gsea-msigdb.org/gsea/msigdb/cards/...  \n",
       "3   http://www.gsea-msigdb.org/gsea/msigdb/cards/...  \n",
       "4   http://www.gsea-msigdb.org/gsea/msigdb/cards/...  \n",
       "\n",
       "[5 rows x 115 columns]"
      ]
     },
     "execution_count": 170,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df = pd.read_csv('/data/g0017139/CNA/enrichment_matrix_Gene Ontology - Biolocal Processes.tsv',sep='\\t')\n",
    "df[['gene_set_name', 'Link']] = df['Unnamed: 0'].str.split(' --',expand=True)\n",
    "df = df.drop('Unnamed: 0', axis=1)\n",
    "df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 171,
   "id": "e619c257",
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "df1 = pd.read_csv('/data/g0017139/CNA/gobp_immune_gene_sets_manualselection_updated.tsv', sep='\\t')\n",
    "df1 = df1[df1['is_immune_process'] == True]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 172,
   "id": "ce57a428",
   "metadata": {},
   "outputs": [],
   "source": [
    "df = pd.merge(df, df1, left_on='gene_set_name', right_on='gene_set_name',how='left')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "4a9cc06d",
   "metadata": {},
   "source": [
    "From GO-BP, all gene set were selected that contained the motif ‘METABOLIC_PROCESS’ in the title. In addition, gene sets that contained the name of a metabolite or class of metabolites in combination with the motif ‘_TRANSPORT’ in the title were selected. Furthermore, gene sets not containing these title motifs, but nevertheless associated with (cancer) metabolism, were manually selected on the basis of metabolic pathway names"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 173,
   "id": "f58ecb3e",
   "metadata": {
    "scrolled": false
   },
   "outputs": [],
   "source": [
    "names = pd.read_csv('/data/g0017139/CNA/metabolic_gene_sets.txt', sep='\\t')['GO_NUCLEOBASE_METABOLIC_PROCESS']\n",
    "df['is_methabolic_process'] = df['gene_set_name'].isin(names.str.replace('GO','GOBP'))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 174,
   "id": "336227a4",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>consensus independent component 1</th>\n",
       "      <th>consensus independent component 2</th>\n",
       "      <th>consensus independent component 3</th>\n",
       "      <th>consensus independent component 4</th>\n",
       "      <th>consensus independent component 5</th>\n",
       "      <th>consensus independent component 6</th>\n",
       "      <th>consensus independent component 7</th>\n",
       "      <th>consensus independent component 8</th>\n",
       "      <th>consensus independent component 9</th>\n",
       "      <th>consensus independent component 10</th>\n",
       "      <th>...</th>\n",
       "      <th>consensus independent component 110</th>\n",
       "      <th>consensus independent component 111</th>\n",
       "      <th>consensus independent component 112</th>\n",
       "      <th>consensus independent component 113</th>\n",
       "      <th>gene_set_name</th>\n",
       "      <th>Link</th>\n",
       "      <th>in_manual_selection</th>\n",
       "      <th>in_immune_terms</th>\n",
       "      <th>is_immune_process</th>\n",
       "      <th>is_methabolic_process</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>-2.782916e-16</td>\n",
       "      <td>1.391458e-16</td>\n",
       "      <td>-2.782916e-16</td>\n",
       "      <td>-2.782916e-16</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>-1.391458e-16</td>\n",
       "      <td>1.768696e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>1.429627e+00</td>\n",
       "      <td>-1.478641</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>GOBP_MITOCHONDRIAL_GENOME_MAINTENANCE</td>\n",
       "      <td>http://www.gsea-msigdb.org/gsea/msigdb/cards/...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>-1.391458e-16</td>\n",
       "      <td>1.505294e+00</td>\n",
       "      <td>1.391458e-16</td>\n",
       "      <td>1.391458e-16</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>-1.391458e-16</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>1.391458e-16</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>GOBP_SINGLE_STRAND_BREAK_REPAIR</td>\n",
       "      <td>http://www.gsea-msigdb.org/gsea/msigdb/cards/...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>-5.668647e-01</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>2.442695e+00</td>\n",
       "      <td>4.174375e-16</td>\n",
       "      <td>2.078227e+00</td>\n",
       "      <td>-1.187087e+00</td>\n",
       "      <td>-1.872795e+00</td>\n",
       "      <td>9.941600e-01</td>\n",
       "      <td>1.610233e+00</td>\n",
       "      <td>...</td>\n",
       "      <td>-5.963215e-01</td>\n",
       "      <td>-1.223743e+00</td>\n",
       "      <td>-0.651827</td>\n",
       "      <td>2.530474e+00</td>\n",
       "      <td>GOBP_REGULATION_OF_DNA_RECOMBINATION</td>\n",
       "      <td>http://www.gsea-msigdb.org/gsea/msigdb/cards/...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1.391458e-16</td>\n",
       "      <td>1.391458e-16</td>\n",
       "      <td>1.391458e-16</td>\n",
       "      <td>-1.391458e-16</td>\n",
       "      <td>-2.782916e-16</td>\n",
       "      <td>-1.391458e-16</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>-1.391458e-16</td>\n",
       "      <td>-1.246121e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>...</td>\n",
       "      <td>1.391458e-16</td>\n",
       "      <td>-1.391458e-16</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>-1.391458e-16</td>\n",
       "      <td>GOBP_RIBOSOMAL_LARGE_SUBUNIT_ASSEMBLY</td>\n",
       "      <td>http://www.gsea-msigdb.org/gsea/msigdb/cards/...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>-1.391458e-16</td>\n",
       "      <td>1.391458e-16</td>\n",
       "      <td>2.782916e-16</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>1.391458e-16</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>...</td>\n",
       "      <td>-1.391458e-16</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>-1.391458e-16</td>\n",
       "      <td>GOBP_RIBOSOMAL_SMALL_SUBUNIT_ASSEMBLY</td>\n",
       "      <td>http://www.gsea-msigdb.org/gsea/msigdb/cards/...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 119 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "   consensus independent component 1  consensus independent component 2  \\\n",
       "0                      -2.782916e-16                       1.391458e-16   \n",
       "1                      -1.391458e-16                       1.505294e+00   \n",
       "2                      -5.668647e-01                       0.000000e+00   \n",
       "3                       1.391458e-16                       1.391458e-16   \n",
       "4                       0.000000e+00                       0.000000e+00   \n",
       "\n",
       "   consensus independent component 3  consensus independent component 4  \\\n",
       "0                      -2.782916e-16                      -2.782916e-16   \n",
       "1                       1.391458e-16                       1.391458e-16   \n",
       "2                       0.000000e+00                       2.442695e+00   \n",
       "3                       1.391458e-16                      -1.391458e-16   \n",
       "4                       0.000000e+00                      -1.391458e-16   \n",
       "\n",
       "   consensus independent component 5  consensus independent component 6  \\\n",
       "0                       0.000000e+00                       0.000000e+00   \n",
       "1                       0.000000e+00                       0.000000e+00   \n",
       "2                       4.174375e-16                       2.078227e+00   \n",
       "3                      -2.782916e-16                      -1.391458e-16   \n",
       "4                       1.391458e-16                       2.782916e-16   \n",
       "\n",
       "   consensus independent component 7  consensus independent component 8  \\\n",
       "0                       0.000000e+00                      -1.391458e-16   \n",
       "1                      -1.391458e-16                       0.000000e+00   \n",
       "2                      -1.187087e+00                      -1.872795e+00   \n",
       "3                       0.000000e+00                      -1.391458e-16   \n",
       "4                       0.000000e+00                       0.000000e+00   \n",
       "\n",
       "   consensus independent component 9  consensus independent component 10  ...  \\\n",
       "0                       1.768696e+00                        0.000000e+00  ...   \n",
       "1                       0.000000e+00                        1.391458e-16  ...   \n",
       "2                       9.941600e-01                        1.610233e+00  ...   \n",
       "3                      -1.246121e+00                        0.000000e+00  ...   \n",
       "4                       1.391458e-16                        0.000000e+00  ...   \n",
       "\n",
       "   consensus independent component 110  consensus independent component 111  \\\n",
       "0                         0.000000e+00                         1.429627e+00   \n",
       "1                         0.000000e+00                         0.000000e+00   \n",
       "2                        -5.963215e-01                        -1.223743e+00   \n",
       "3                         1.391458e-16                        -1.391458e-16   \n",
       "4                        -1.391458e-16                         0.000000e+00   \n",
       "\n",
       "   consensus independent component 112  consensus independent component 113  \\\n",
       "0                            -1.478641                         0.000000e+00   \n",
       "1                             0.000000                         0.000000e+00   \n",
       "2                            -0.651827                         2.530474e+00   \n",
       "3                             0.000000                        -1.391458e-16   \n",
       "4                             0.000000                        -1.391458e-16   \n",
       "\n",
       "                           gene_set_name  \\\n",
       "0  GOBP_MITOCHONDRIAL_GENOME_MAINTENANCE   \n",
       "1        GOBP_SINGLE_STRAND_BREAK_REPAIR   \n",
       "2   GOBP_REGULATION_OF_DNA_RECOMBINATION   \n",
       "3  GOBP_RIBOSOMAL_LARGE_SUBUNIT_ASSEMBLY   \n",
       "4  GOBP_RIBOSOMAL_SMALL_SUBUNIT_ASSEMBLY   \n",
       "\n",
       "                                                Link  in_manual_selection  \\\n",
       "0   http://www.gsea-msigdb.org/gsea/msigdb/cards/...                  NaN   \n",
       "1   http://www.gsea-msigdb.org/gsea/msigdb/cards/...                  NaN   \n",
       "2   http://www.gsea-msigdb.org/gsea/msigdb/cards/...                  NaN   \n",
       "3   http://www.gsea-msigdb.org/gsea/msigdb/cards/...                  NaN   \n",
       "4   http://www.gsea-msigdb.org/gsea/msigdb/cards/...                  NaN   \n",
       "\n",
       "   in_immune_terms  is_immune_process  is_methabolic_process  \n",
       "0              NaN                NaN                  False  \n",
       "1              NaN                NaN                  False  \n",
       "2              NaN                NaN                  False  \n",
       "3              NaN                NaN                  False  \n",
       "4              NaN                NaN                  False  \n",
       "\n",
       "[5 rows x 119 columns]"
      ]
     },
     "execution_count": 174,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 175,
   "id": "4e63ec46",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "is_methabolic_process = 267\n",
      "is_immune_process = 675\n",
      "total = 5030\n"
     ]
    }
   ],
   "source": [
    "print(f'is_methabolic_process = {df[\"is_methabolic_process\"].sum()}')\n",
    "print(f'is_immune_process = {df[\"is_immune_process\"].sum()}')\n",
    "print(f'total = {len(df)}')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 182,
   "id": "3e7ce5ba",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0.05308151093439364"
      ]
     },
     "execution_count": 182,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "267/5030"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 176,
   "id": "82a92ab8",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(1238, 115)"
      ]
     },
     "execution_count": 176,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df = pd.read_csv('/data/g0017139/CNA/enrichment_matrix_Reactome.tsv',sep='\\t')\n",
    "df[['gene_set_name', 'Link']] = df['Unnamed: 0'].str.split(' --',expand=True)\n",
    "df = df.drop('Unnamed: 0', axis=1)\n",
    "df.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 177,
   "id": "716454a6",
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "df1 = pd.read_csv('/data/g0017139/CNA/reactome_immune_gene_sets.tsv', sep='\\t')\n",
    "df1 = df1[df1['is_immune_process'] == True]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 178,
   "id": "fdec2a34",
   "metadata": {},
   "outputs": [],
   "source": [
    "df = pd.merge(df, df1, left_on='gene_set_name', right_on='gene_set_name',how='left')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "e3c87060",
   "metadata": {},
   "source": [
    "From GO-BP, all gene set were selected that contained the motif ‘METABOLIC_PROCESS’ in the title. In addition, gene sets that contained the name of a metabolite or class of metabolites in combination with the motif ‘_TRANSPORT’ in the title were selected. Furthermore, gene sets not containing these title motifs, but nevertheless associated with (cancer) metabolism, were manually selected on the basis of metabolic pathway names"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 179,
   "id": "4694fdc3",
   "metadata": {
    "scrolled": false
   },
   "outputs": [],
   "source": [
    "df['is_methabolic_process'] = df['gene_set_name'].isin(names)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 180,
   "id": "a9b4f914",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>consensus independent component 1</th>\n",
       "      <th>consensus independent component 2</th>\n",
       "      <th>consensus independent component 3</th>\n",
       "      <th>consensus independent component 4</th>\n",
       "      <th>consensus independent component 5</th>\n",
       "      <th>consensus independent component 6</th>\n",
       "      <th>consensus independent component 7</th>\n",
       "      <th>consensus independent component 8</th>\n",
       "      <th>consensus independent component 9</th>\n",
       "      <th>consensus independent component 10</th>\n",
       "      <th>...</th>\n",
       "      <th>consensus independent component 108</th>\n",
       "      <th>consensus independent component 109</th>\n",
       "      <th>consensus independent component 110</th>\n",
       "      <th>consensus independent component 111</th>\n",
       "      <th>consensus independent component 112</th>\n",
       "      <th>consensus independent component 113</th>\n",
       "      <th>gene_set_name</th>\n",
       "      <th>Link</th>\n",
       "      <th>is_immune_process</th>\n",
       "      <th>is_methabolic_process</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>-1.391458e-16</td>\n",
       "      <td>1.391458e-16</td>\n",
       "      <td>1.391458e-16</td>\n",
       "      <td>1.391458e-16</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>-1.391458e-16</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>1.391458e-16</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>-1.391458e-16</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>REACTOME_INTERLEUKIN_6_SIGNALING</td>\n",
       "      <td>http://www.gsea-msigdb.org/gsea/msigdb/cards/...</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>-8.247059e-01</td>\n",
       "      <td>-1.391458e-16</td>\n",
       "      <td>9.053910e-01</td>\n",
       "      <td>-2.850296e+00</td>\n",
       "      <td>-2.782916e-16</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>8.057245e-01</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>...</td>\n",
       "      <td>7.504103e-01</td>\n",
       "      <td>4.598313e-01</td>\n",
       "      <td>6.413760e-01</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>-4.672263e-01</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>REACTOME_APOPTOSIS</td>\n",
       "      <td>http://www.gsea-msigdb.org/gsea/msigdb/cards/...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>1.514928e+00</td>\n",
       "      <td>-1.312150e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>-1.391458e-16</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>-1.391458e-16</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>2.782916e-16</td>\n",
       "      <td>...</td>\n",
       "      <td>1.370731e+00</td>\n",
       "      <td>2.782916e-16</td>\n",
       "      <td>-9.286997e-01</td>\n",
       "      <td>9.296210e-01</td>\n",
       "      <td>1.391458e-16</td>\n",
       "      <td>-1.391458e-16</td>\n",
       "      <td>REACTOME_INTRINSIC_PATHWAY_FOR_APOPTOSIS</td>\n",
       "      <td>http://www.gsea-msigdb.org/gsea/msigdb/cards/...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1.391458e-16</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>2.235452e+00</td>\n",
       "      <td>1.391458e-16</td>\n",
       "      <td>2.782916e-16</td>\n",
       "      <td>-1.391458e-16</td>\n",
       "      <td>1.391458e-16</td>\n",
       "      <td>1.391458e-16</td>\n",
       "      <td>1.391458e-16</td>\n",
       "      <td>-1.391458e-16</td>\n",
       "      <td>...</td>\n",
       "      <td>1.391458e-16</td>\n",
       "      <td>-1.391458e-16</td>\n",
       "      <td>1.391458e-16</td>\n",
       "      <td>-1.391458e-16</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>-1.391458e-16</td>\n",
       "      <td>REACTOME_MAPK3_ERK1_ACTIVATION</td>\n",
       "      <td>http://www.gsea-msigdb.org/gsea/msigdb/cards/...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>-1.391458e-16</td>\n",
       "      <td>1.701075e+00</td>\n",
       "      <td>-1.391458e-16</td>\n",
       "      <td>-3.037907e+00</td>\n",
       "      <td>-1.892171e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>-1.391458e-16</td>\n",
       "      <td>1.391458e-16</td>\n",
       "      <td>-2.782916e-16</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>-2.782916e-16</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>1.391458e-16</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>REACTOME_TRANSLESION_SYNTHESIS_BY_Y_FAMILY_DNA...</td>\n",
       "      <td>http://www.gsea-msigdb.org/gsea/msigdb/cards/...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 117 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "   consensus independent component 1  consensus independent component 2  \\\n",
       "0                      -1.391458e-16                       1.391458e-16   \n",
       "1                      -8.247059e-01                      -1.391458e-16   \n",
       "2                       0.000000e+00                       0.000000e+00   \n",
       "3                       1.391458e-16                       0.000000e+00   \n",
       "4                      -1.391458e-16                       1.701075e+00   \n",
       "\n",
       "   consensus independent component 3  consensus independent component 4  \\\n",
       "0                       1.391458e-16                       1.391458e-16   \n",
       "1                       9.053910e-01                      -2.850296e+00   \n",
       "2                       1.514928e+00                      -1.312150e+00   \n",
       "3                       2.235452e+00                       1.391458e-16   \n",
       "4                      -1.391458e-16                      -3.037907e+00   \n",
       "\n",
       "   consensus independent component 5  consensus independent component 6  \\\n",
       "0                       0.000000e+00                       0.000000e+00   \n",
       "1                      -2.782916e-16                       0.000000e+00   \n",
       "2                       0.000000e+00                      -1.391458e-16   \n",
       "3                       2.782916e-16                      -1.391458e-16   \n",
       "4                      -1.892171e+00                       0.000000e+00   \n",
       "\n",
       "   consensus independent component 7  consensus independent component 8  \\\n",
       "0                      -1.391458e-16                       0.000000e+00   \n",
       "1                       0.000000e+00                       8.057245e-01   \n",
       "2                       0.000000e+00                      -1.391458e-16   \n",
       "3                       1.391458e-16                       1.391458e-16   \n",
       "4                      -1.391458e-16                       1.391458e-16   \n",
       "\n",
       "   consensus independent component 9  consensus independent component 10  ...  \\\n",
       "0                       0.000000e+00                        1.391458e-16  ...   \n",
       "1                       0.000000e+00                        0.000000e+00  ...   \n",
       "2                       0.000000e+00                        2.782916e-16  ...   \n",
       "3                       1.391458e-16                       -1.391458e-16  ...   \n",
       "4                      -2.782916e-16                        0.000000e+00  ...   \n",
       "\n",
       "   consensus independent component 108  consensus independent component 109  \\\n",
       "0                         0.000000e+00                        -1.391458e-16   \n",
       "1                         7.504103e-01                         4.598313e-01   \n",
       "2                         1.370731e+00                         2.782916e-16   \n",
       "3                         1.391458e-16                        -1.391458e-16   \n",
       "4                         0.000000e+00                        -2.782916e-16   \n",
       "\n",
       "   consensus independent component 110  consensus independent component 111  \\\n",
       "0                         0.000000e+00                         0.000000e+00   \n",
       "1                         6.413760e-01                         0.000000e+00   \n",
       "2                        -9.286997e-01                         9.296210e-01   \n",
       "3                         1.391458e-16                        -1.391458e-16   \n",
       "4                         0.000000e+00                         0.000000e+00   \n",
       "\n",
       "   consensus independent component 112  consensus independent component 113  \\\n",
       "0                         0.000000e+00                         0.000000e+00   \n",
       "1                        -4.672263e-01                         0.000000e+00   \n",
       "2                         1.391458e-16                        -1.391458e-16   \n",
       "3                         0.000000e+00                        -1.391458e-16   \n",
       "4                         1.391458e-16                         0.000000e+00   \n",
       "\n",
       "                                       gene_set_name  \\\n",
       "0                   REACTOME_INTERLEUKIN_6_SIGNALING   \n",
       "1                                 REACTOME_APOPTOSIS   \n",
       "2           REACTOME_INTRINSIC_PATHWAY_FOR_APOPTOSIS   \n",
       "3                     REACTOME_MAPK3_ERK1_ACTIVATION   \n",
       "4  REACTOME_TRANSLESION_SYNTHESIS_BY_Y_FAMILY_DNA...   \n",
       "\n",
       "                                                Link  is_immune_process  \\\n",
       "0   http://www.gsea-msigdb.org/gsea/msigdb/cards/...               True   \n",
       "1   http://www.gsea-msigdb.org/gsea/msigdb/cards/...                NaN   \n",
       "2   http://www.gsea-msigdb.org/gsea/msigdb/cards/...                NaN   \n",
       "3   http://www.gsea-msigdb.org/gsea/msigdb/cards/...                NaN   \n",
       "4   http://www.gsea-msigdb.org/gsea/msigdb/cards/...                NaN   \n",
       "\n",
       "   is_methabolic_process  \n",
       "0                  False  \n",
       "1                  False  \n",
       "2                  False  \n",
       "3                  False  \n",
       "4                  False  \n",
       "\n",
       "[5 rows x 117 columns]"
      ]
     },
     "execution_count": 180,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 181,
   "id": "a485b8d9",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "is_methabolic_process = 26\n",
      "is_immune_process = 73\n",
      "total = 1238\n"
     ]
    }
   ],
   "source": [
    "print(f'is_methabolic_process = {df[\"is_methabolic_process\"].sum()}')\n",
    "print(f'is_immune_process = {df[\"is_immune_process\"].sum()}')\n",
    "print(f'total = {len(df)}')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "707e03d5",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "TryTwo",
   "language": "python",
   "name": "trytwo"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.6"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
